From 028c4416216af21dcae53f7daa4f549d0c2bad4f Mon Sep 17 00:00:00 2001
From: James Simmons <uja.ornl@yahoo.com>
Date: Fri, 3 Jun 2016 14:09:54 -0400
Subject: [PATCH 7/8] LU-8056 lloop: fix bio_for_each_segment_all for newer
 kernels

Lustre patch http://review.whamcloud.com/20478 back ported
bio_for_each_segment_all from newer kernels but support
for newer kernels was done incorrectly. Update the code to
work with newer kernels.

Change-Id: I6a926320f80113169a13d2319190721c83d58b1d
Signed-off-by: James Simmons <uja.ornl@yahoo.com>
---
 lustre/llite/lloop.c | 15 +--------------
 1 file changed, 1 insertion(+), 14 deletions(-)

diff --git a/lustre/llite/lloop.c b/lustre/llite/lloop.c
index 64318ca..64da8c2 100644
--- a/lustre/llite/lloop.c
+++ b/lustre/llite/lloop.c
@@ -193,13 +193,8 @@ static int do_bio_lustrebacked(struct lloop_device *lo, struct bio *head)
         struct cl_object     *obj = ll_i2info(inode)->lli_clob;
         pgoff_t               offset;
         int                   ret;
-#ifdef HAVE_BVEC_ITER
-	struct bvec_iter      iter;
-	struct bio_vec        bvec;
-#else
 	int		      iter;
 	struct bio_vec	     *bvec;
-#endif
         int                   rw;
 	size_t		      page_count = 0;
         struct bio           *bio;
@@ -226,16 +221,9 @@ static int do_bio_lustrebacked(struct lloop_device *lo, struct bio *head)
 
 #ifdef HAVE_BVEC_ITER
 		offset = (pgoff_t)(bio->bi_iter.bi_sector << 9) + lo->lo_offset;
-		bio_for_each_segment_all(bvec, bio, iter) {
-			BUG_ON(bvec.bv_offset != 0);
-			BUG_ON(bvec.bv_len != PAGE_CACHE_SIZE);
-
-			pages[page_count] = bvec.bv_page;
-			offsets[page_count] = offset;
-			page_count++;
-			offset += bvec.bv_len;
 #else
 		offset = (pgoff_t)(bio->bi_sector << 9) + lo->lo_offset;
+#endif
 		bio_for_each_segment_all(bvec, bio, iter) {
 			BUG_ON(bvec->bv_offset != 0);
 			BUG_ON(bvec->bv_len != PAGE_CACHE_SIZE);
@@ -244,7 +232,6 @@ static int do_bio_lustrebacked(struct lloop_device *lo, struct bio *head)
 			offsets[page_count] = offset;
 			page_count++;
 			offset += bvec->bv_len;
-#endif
 		}
 		LASSERT(page_count <= LLOOP_MAX_SEGMENTS);
 	}
-- 
2.8.2

